### PR TITLE
feat: add error toggle

### PR DESCRIPTION
**Pull request by Blackman99**
### Motivation
When focused on coding the error message might be an obstacle in the editor. 
So maybe it is better to add a control to let the user decide whether to show the message or not. 

### Changes

* Fix the `codeType` in `CodeMirrorEditor`
* Add `MessageToggle` component
* Add `showMessage` ref and use `MessageToggle` in `EditorContaner` to control `Message` render

### Screenshots
add-message-toggles

This PR can possibly solve #64

### CODE REVIEW

**Comment by yyx990803** [comment]
Thanks for the PR - we just merged a major PR that lands Monaco editor support which seems to have caused some conflict. Would you mind rebasing your PR?

**Comment by Blackman99** [comment]
Oh, sure thing!

**Comment by Blackman99** [comment]
Rebase done. I noticed that the `codeType` in CodeMirrorEditor is wrong. I also added the fix code. Hope it is OK

**Comment by yyx990803** [comment]
I tried it, but not sure if the top-right corner is the right place to put the toggle. It's quite far from the messages themselves so the connection is not obvious.

Maybe it should be floating in the bottom-left corner of the editor - so when the user dismisses an error when they don't want it, they can instantly find the toggle to turn it off.

**Comment by Blackman99** [comment]
Indeed 
And I've tried placing the toggle at the bottom-left corner of the editor. It would 
cover up some of the message content. 
So I move it to the bottom-right corner.
Right now it looks like this:

add-message-toggles

File: src/editor/MessageToggle.vue [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<script setup lang="ts">
```
Comment (sxzz) [inline src/editor/MessageToggle.vue:1]: Please format the code `pnpm run format`

File: src/editor/MessageToggle.vue [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  withDefaults(defineProps<{
```
Comment (sxzz) [inline src/editor/MessageToggle.vue:2]: We can use `defineModel`

File: src/editor/EditorContainer.vue [hunk 3]
```diff
@@ -26,9 +28,10 @@ const onChange = debounce((code: string) => {
       :filename="store.state.activeFile.filename"
     />
     <Message
-      v-if="props.editorComponent.editorType !== 'monaco'"
+      v-if="props.editorComponent.editorType !== 'monaco' && showMessage"
       :err="store.state.errors[0]"
     />
+    <MessageToggle v-model="showMessage" />
```
Comment (sxzz) [inline src/editor/EditorContainer.vue:34]: I think if it's not monaco, we can hide the toggle.

File: src/editor/MessageToggle.vue (hunk 1 above)
Comment (Blackman99) [reply src/editor/MessageToggle.vue:1]: Done
But there are two problems with prettier
* prettier is not in the package.json as devDependencies
* `pnpm run format` will format all files. It is better to format the changed files only by running `prettier path/to/changed/file --write`

File: src/editor/MessageToggle.vue (hunk 2 above)
Comment (Blackman99) [reply src/editor/MessageToggle.vue]: Done

File: src/editor/EditorContainer.vue (hunk 3 above)
Comment (Blackman99) [reply src/editor/EditorContainer.vue]: Done

File: src/editor/MessageToggle.vue [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  })
+
+  defineEmits<{
+    (e: 'update:modelValue', next: boolean): void 
+  }>()
+</script>
+<template>
+  <div class="wrapper" @click="$emit('update:modelValue', !modelValue)">
+    <span>
+      Message
```
[the remark this answers is not in the document]
Comment (Blackman99) [reply src/editor/MessageToggle.vue]: Indeed, Done

File: src/editor/MessageToggle.vue [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  }
+)
+
+defineEmits<{
+  (e: 'update:modelValue', next: boolean): void
+}>()
+</script>
+<template>
+  <div class="wrapper" @click="$emit('update:modelValue', !modelValue)">
+    <span> Show Error </span>
```
Comment (sxzz) [inline src/editor/MessageToggle.vue:17]:
```suggestion
    <span>Show Error</span>
```

File: src/editor/EditorContainer.vue [hunk 6]
```diff
@@ -25,10 +27,13 @@ const onChange = debounce((code: string) => {
       :value="store.state.activeFile.code"
       :filename="store.state.activeFile.filename"
     />
-    <Message
-      v-if="props.editorComponent.editorType !== 'monaco'"
-      :err="store.state.errors[0]"
-    />
+    <template v-if="editorComponent.editorType !== 'monaco'">
+      <Message
+        v-if="showMessage"
```
Comment (sxzz) [inline src/editor/EditorContainer.vue:32]:
```suggestion
        v-show="showMessage"
```

File: src/editor/MessageToggle.vue (hunk 5 above)
Comment (Blackman99) [reply src/editor/MessageToggle.vue]: Done

File: src/editor/EditorContainer.vue (hunk 6 above)
Comment (Blackman99) [reply src/editor/EditorContainer.vue]: Done

**Review by sxzz — APPROVED** [review]

File: src/editor/MessageToggle.vue [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  defineProps<{
+    modelValue?: boolean
+  }>(),
+  {
+    modelValue: false,
+  }
+)
+
+defineEmits<{
+  (e: 'update:modelValue', next: boolean): void
```
Comment (sxzz) [inline src/editor/MessageToggle.vue:12]:
```suggestion
  (e: 'update:modelValue', value: boolean): void
```

**Pull request merged**